backend/src/graphql/resolver/ContributionResolver.test.ts

```diff
@@ -121,4 +125,114 @@ describe('ContributionResolver', () => {
       })
     })
   })
+
+  describe('listContributions', () => {
+    describe('unauthenticated', () => {
+      it('returns an error', async () => {
+        await expect(
+          query({
+            query: listContributions,
+            variables: {
+              currentPage: 1,
+              pageSize: 25,
+              order: 'DESC',
+              filterConfirmed: false,
+            },
+          }),
+        ).resolves.toEqual(
+          expect.objectContaining({
+            errors: [new GraphQLError('401 Unauthorized')],
+          }),
+        )
+      })
+    })
+
+    describe('authenticated', () => {
+      beforeAll(async () => {
+        await userFactory(testEnv, bibiBloxberg)
+        await query({
+          query: login,
+          variables: { email: 'bibi@bloxberg.de', password: 'Aa12345_' },
+        })
+        await mutate({
+          mutation: createContribution,
+          variables: {
+            amount: 100.0,
+            memo: 'Test env contribution',
+            creationDate: new Date().toString(),
+          },
+        })
+      })
+
+      it('returns only unconfirmed creations', async () => {
+        await expect(
+          query({
+            query: listContributions,
+            variables: {
+              currentPage: 1,
+              pageSize: 25,
+              order: 'DESC',
+              filterConfirmed: true,
+            },
+          }),
+        ).resolves.toEqual(
+          expect.objectContaining({
+            data: {
+              listContributions: expect.arrayContaining([
+                expect.objectContaining({
+                  id: expect.any(Number),
+                  memo: 'Test env contribution',
+                  amount: '100',
+                }),
+              ]),
+            },
+          }),
+        )
+      })
+
+      describe('Adding confirmed creations', () => {
+        beforeAll(async () => {
+          await userFactory(testEnv, peterLustig)
+          const bibisCreation = creations.find((creation) => creation.email === 'bibi@bloxberg.de')
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          await creationFactory(testEnv, bibisCreation!)
+          await query({
+            query: login,
+            variables: { email: 'bibi@bloxberg.de', password: 'Aa12345_' },
+          })
+        })
+
+        it('filter confirmed is false', async () => {
```

Comment:
As I said before, I want a describe block for the value of the filter, and another test with the filter enabled.

Reply:
Change it so that when we go in authenticated we set-up an admin confirmed creation and a user not confirmed creation.
After that we have the 2 describe blocks and the it blocks in it!
Hope it is now like you imagined it!?